File: frontend/services/studentService.js
import api from "./api";

export const getStudents = (params) => {
  return api.get("/api/students", {
    params: {
      page: params.page,
      size: params.size,
      search: params.search,
      sortBy: params.sortBy,
      sortDir: params.sortDir,
    },
  });
};

export const createStudent = (data) => {
  return api.post("/api/students", data);
};

export const updateStudent = (id, data) => {
  return api.put(`/api/students/${id}`, data);
};

export const deleteStudent = (id) => {
  return api.delete(`/api/students/${id}`);
};

File: Backend/src/main/java/com/assignment/courseManagementSystem/services/contract/StudentService.java
package com.assignment.courseManagementSystem.services.contract;

import com.assignment.courseManagementSystem.dto.StudentDto;
import org.springframework.data.domain.Page;

public interface StudentService {

    StudentDto createStudent(StudentDto dto);

    StudentDto getStudentById(Long id);

    Page<StudentDto> getStudents(
            String search,
            int page,
            int size,
            String sortBy,
            String sortDir
    );

    StudentDto updateStudent(Long id, StudentDto dto);

    void deleteStudent(Long id);
}


